source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -157,6 +157,13 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
         !SSL_CTX_set1_curves_list(ctx.ssl_ctx_.get(), config.ecdhCurves().c_str())) {
       throw EnvoyException(absl::StrCat("Failed to initialize ECDH curves ", config.ecdhCurves()));
     }
+
+    // Set signature algorithms if given, otherwise fall back to BoringSSL defaults.
+    if (!capabilities_.provides_sigalgs && !config.signatureAlgorithms().empty() &&
```

Comment:
nit: for readability, break up the condition for whether to call `SSL_CTX_set1_sigalgs_list` from the error handling for a failure in that call.